### PR TITLE
DT-6532:  In vehicle cards

### PR DESCRIPTION
**Pull request by Antiik91**
## Proposed Changes

  - Shows in transit information in navigator
  - 

## Pull Request Check List

  - A reasonable set of unit tests is included
  - Console does not show new warnings/errors
  - Changes are documented or they are self explanatory
  - This pull request does not have any merge conflicts
  - All existing tests pass in CI build

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform
  - Check that the implementation matches the design, when such one is defined in a Jira issue
  - Merge the pull request

### CODE REVIEW

File: app/component/itinerary/NaviCardExtension.js
Comment (partisaani) [inline app/component/itinerary/NaviCardExtension.js:1]: DRY and KISS; I would refactor the two embedded return values as individual components for reduced complexity of this parent component
```jsx
return (
  <div className="extension">
    <div className="extension-divider" />
    {legType === 'in-transit' ? <TransitLegExtender {...props} /> : <WalkLegExtender {...props} />}
  </div>
)
```
also simplifies testing for all of the three components as for parent you can just sinon.stub() the `*Extender` components with an easily locatable attribute and for the `*Extender`s you can just make sure the component looks as it should with given props

**Review by partisaani — CHANGES_REQUESTED** [review]

File: app/component/itinerary/NaviCardExtension.js
Comment (Antiik91) [reply app/component/itinerary/NaviCardExtension.js:1]: Good point, although I reckon that we get a lot of navigation components before we are finished with this, so we don't want to spread too thin here.

I'll take another opinion here as well before I start to refactor this.

File: app/component/itinerary/NaviCardContainer.js [hunk 1]
```diff
@@ -151,27 +151,26 @@ function NaviCardContainer(
...
-          leg={currentLeg}
-          nextLeg={nextLeg}
-          cardExpanded={cardExpanded}
-          legType={legType}
-        />
-      );
     } else {
-      naviTopContent = `Tracking ${currentLeg?.mode} leg`;
+      legType = 'in-transit';
     }
```
Comment (vesameskanen) [inline app/component/itinerary/NaviCardContainer.js]: Maybe handle leg types like this:

```
export const LEGTYPE = {
  WAIT: 'WAIT',
  MOVE: 'MOVE',
  TRANSIT: 'TRANSIT',
}
```

Add this to  naviUtils and import it from there to other classes which refer to leg types.

File: app/component/itinerary/NaviCardContainer.js [hunk 2]
```diff
@@ -151,27 +151,26 @@ function NaviCardContainer(
...
+      <NaviCard
+        leg={currentLeg}
+        nextLeg={nextLeg}
+        cardExpanded={cardExpanded}
+        legType={legType}
+      />
+    );
   } else if (time > legTime(last.end)) {
     naviTopContent = <FormattedMessage id="navigation-journey-end" />;
   } else {
```
Comment (vesameskanen) [inline app/component/itinerary/NaviCardContainer.js:177]: There are now 2 separate ways to render wait state. Maybe these formatted messages could be moved to NaviCard? In other words,  leg would not be required prop and legType could be also `ITINERARY_PENDING` and `ITINERARY_ENDED` or something like that.

File: app/component/itinerary/NaviCard.js [hunk 3]
```diff
@@ -12,13 +12,19 @@ const iconMap = {
...
   WAIT: 'icon-icon_navigation_wait',
+  BUS: 'icon-icon_bus',
+  RAIL: 'icon-icon_rail',
+  SUBWAY: 'icon-icon_subway',
+  TRAM: 'icon-icon_tram',
+  FERRY: 'icon-icon_ferry',
 };
 
 export default function NaviCard({ leg, nextLeg, legType, cardExpanded }) {
   const iconName = legType === 'wait' ? iconMap.WAIT : iconMap[leg.mode];
```
Comment (vesameskanen) [inline app/component/itinerary/NaviCard.js]: nextLeg is not a required prop, can be undefined

File: app/component/itinerary/NaviCardExtension.js
Comment (vesameskanen) [reply app/component/itinerary/NaviCardExtension.js:1]: I don't have a stong opinion about this. I guess we could put all this navi stuff to a sub folder first, and after that start splitting into smaller components.

File: app/component/itinerary/NaviCardExtension.js [hunk 4]
```diff
@@ -21,10 +25,51 @@ const NaviCardExtension = ({ leg }, { config }) => {
...
+    const now = new Date();
+    const idx =
+      intermediatePlaces.findIndex(
+        p =>
+          new Date(
+            p.arrival.estimated?.time || p.arrival.scheduledTime,
+          ).getTime() > now.getTime(),
+      ) ?? -1;
+
+    const count = idx > -1 ? intermediatePlaces.length - idx : 0;
```
Comment (vesameskanen) [inline app/component/itinerary/NaviCardExtension.js]: Use Date.now() and legTime.

File: app/component/itinerary/NaviCardExtension.js [hunk 5]
```diff
@@ -21,10 +25,51 @@ const NaviCardExtension = ({ leg }, { config }) => {
...
+    const stopCountElement = <span className="realtime"> {count}</span>;
```
Comment (vesameskanen) [inline app/component/itinerary/NaviCardExtension.js]: I guess plain `stopCount` isa bit  simpler. We don't have a convention to name jsx snippets as elements.

File: app/component/itinerary/NaviStack.js [hunk 6]
```diff
@@ -29,10 +39,12 @@ NaviStack.propTypes = {
   ).isRequired,
   handleRemove: PropTypes.func.isRequired,
   cardExpanded: PropTypes.bool,
+  legType: PropTypes.string,
 };
 
 NaviStack.defaultProps = {
   cardExpanded: false,
+  legType: '',
```
Comment (vesameskanen) [inline app/component/itinerary/NaviStack.js]: legType does not have logical connection to NaviStack which may contain messages for many legs. I suggest passing the class postfix itself, or maybe handling the vertical offset completely in parent component.

**Review by vesameskanen — CHANGES_REQUESTED** [review]

File: app/component/itinerary/NaviCardExtension.js
Comment (Antiik91) [reply app/component/itinerary/NaviCardExtension.js:1]: I'll create the subfolder in this pr. And after I start refactoring the wait leg, I'll get back to this.

File: app/component/itinerary/NaviCardContainer.js (hunk 2 above)
Comment (Antiik91) [reply app/component/itinerary/NaviCardContainer.js:177]: This will be considered in the next PR, where I'll refactor the wait leg

**Review by vesameskanen — APPROVED** [review]

**Pull request merged**